Sources/SwiftPlot/Histogram.swift

```diff
@@ -276,111 +276,79 @@ extension Histogram: HasGraphLayout {
             }
         }
     }
-
-    //functions to draw the plot
+    
+    /// Draw with rectangles if `histogramType` is `.bar` or with lines if `histogramType` is `.step`
     public func drawData(markers: PlotMarkers, size: Size, renderer: Renderer) {
-        var xM = Float(xMargin)
+        let binCount = histogramSeries.bins
+        let allSeries = [histogramSeries] + histogramStackSeries
         switch histogramSeries.histogramSeriesOptions.histogramType {
         case .bar:
-            for i in 0..<histogramSeries.bins {
-                var currentHeight: Float = histogramSeries.scaledBinFrequency[i]
-                var rect = Rect(
-                    origin: Point(xM, 0),
-                    size: Size(width: barWidth, height: currentHeight)
-                )
-                renderer.drawSolidRect(rect,
-                                       fillColor: histogramSeries.color,
-                                       hatchPattern: .none)
-
-                for series in histogramStackSeries {
-                    rect.origin.y = currentHeight
-                    rect.size.height = series.scaledBinFrequency[i]
-                    renderer.drawSolidRect(rect,
-                                           fillColor: series.color,
-                                           hatchPattern: .none)
-                    currentHeight += series.scaledBinFrequency[i]
+            // Generate all rects needed
+            let xStart = Float(xMargin)
+            let xValues = stride(from: xStart, to: xStart + Float(binCount) * barWidth, by: barWidth)
+            var rects: [Rect] = xValues.map { Rect(origin: Point($0, 0.0), size: Size(width: barWidth, height: 0.0)) }
```

Comment:
It seems like we should be able to calculate these in the loop, rather than creating an extra array for them.

Reply:
hum the only thing I can think of is that we could be constructing the rects in the loop just as needed, but I think we still need an array to store the heights nonetheless, so that we can accumulate the height of all the columns. Am I missing something? I will change it so we only create an array of `Float`s instead for now.

Reply:
At least for the `bar` style, we can construct it bar-by-bar, rather than series-by-series. Then we only need to keep track of the current bar's height.

Reply:
oh you're right, I wasn't thinking straight.

Reply:
Is this better or is there a simpler way?
```
...
case .bar:
    let xStart = Float(xMargin)
    let xValues = stride(from: xStart, to: xStart + Float(binCount) * barWidth, by: barWidth)
    
    // Get a `Slice` of frequencies for each series so we can take one element from each series for
each x value
    var frequencySlices = allSeries.map { $0.scaledBinFrequency[...] }
    var currentHeight: Float = 0.0
    for x in xValues {
        for (series, index) in zip(allSeries, frequencySlices.indices) {
            let height = frequencySlices[index].removeFirst()
            let rect = Rect(origin: Point(x, currentHeight), size: Size(width: barWidth, height:
height))
            renderer.drawSolidRect(rect, fillColor: series.color, hatchPattern: .none)
            currentHeight = height
        }
        currentHeight = 0.0
    }
...
```

Reply:
Sure, looks fine... I think.